satpy/readers/cmsaf_claas2.py

```diff
@@ -74,11 +95,14 @@ def get_dataset(self, dataset_id, info):
 
     def get_area_def(self, dataset_id):
         """Get the area definition."""
-        return pyresample.geometry.AreaDefinition(
-                "some_area_name",
-                "on-the-fly area",
-                "geos",
-                self["/attr/CMSAF_proj4_params"],
-                self["/dimension/x"],
-                self["/dimension/y"],
-                self["/attr/CMSAF_area_extent"])
+        full_disk = self._get_full_disk()
```

Comment:
I would put this in `_subset_full_disk`, what do you think?

Reply:
Good idea, thanks!